backend/controller/observability/calls.go

```diff
@@ -0,0 +1,133 @@
+package observability
+
+import (
+	"context"
+	"github.com/TBD54566975/ftl/backend/controller/dal"
+	ftlv1 "github.com/TBD54566975/ftl/backend/protos/xyz/block/ftl/v1"
+	"github.com/TBD54566975/ftl/backend/schema"
+	"github.com/TBD54566975/ftl/internal/log"
+	"github.com/TBD54566975/ftl/internal/model"
+	"github.com/alecthomas/types/optional"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+	"sync"
+	"time"
+)
+
+var callMeter = otel.Meter("ftl.call")
+var callInitOnce = sync.Once{}
+
+var callRequests metric.Int64Counter
+var callFailures metric.Int64Counter
+var callActive metric.Int64UpDownCounter
+var callLatency metric.Int64Histogram
+
+type CallBegin struct {
+	DestVerb *schema.Ref
+	Callers  []*schema.Ref
+}
+
+type CallEnd struct {
+	DeploymentKey model.DeploymentKey
+	RequestKey    model.RequestKey
+	StartTime     time.Time
+	DestVerb      *schema.Ref
+	Callers       []*schema.Ref
+	Request       *ftlv1.CallRequest
+	Response      optional.Option[*ftlv1.CallResponse]
+	CallError     optional.Option[error]
+}
+
+func initCallMetrics() {
+	callInitOnce.Do(func() {
+		callRequests, _ = callMeter.Int64Counter("ftl.call.requests",
+			metric.WithDescription("number of verb calls"),
+			metric.WithUnit("{count}"))
+		callFailures, _ = callMeter.Int64Counter("ftl.call.failures",
+			metric.WithDescription("number of verb call failures"),
+			metric.WithUnit("{count}"))
+		callActive, _ = callMeter.Int64UpDownCounter("ftl.call.active",
+			metric.WithDescription("number of in flight calls"),
+			metric.WithUnit("{count}"))
+		callLatency, _ = callMeter.Int64Histogram("ftl.call.latency",
+			metric.WithDescription("verb call latency"),
+			metric.WithUnit("{ms}"))
+	})
+}
+
+func RecordCallBegin(ctx context.Context, call *CallBegin) {
+	initCallMetrics()
```

Comment:
I think I'm missing the context on why we need to have an `init` in these metric funcs. It would be awesome if they were just plain old functions that could define the metric and attributes, without having to initialize the other structures. It might also remove the need for `sync.Once` code as well. Maybe we can sync on why the init stuff is required.

Reply:
The otel getting started guides pre-initialize their metrics and use the initialized metrics in their instrumentation. The initialization metric process is a black box (for me at least) so my preference is to avoid risking introducing heavy weight operations in instrumentation code

Reply:
Gotcha! I'm cool with that approach too. I might suggest having like a `New` function on these and maybe an overall `New` that calls the individual "feature" metric files. Then we can just `New` it up when we need it vs. having to have these init funcs everywhere.